code/app/src/main/java/ca/ualberta/ishelf/BookProfileActivity.java

```diff
@@ -541,4 +550,14 @@ public void Booking(View v){
 
     }
 
+    public void MapButton(View v){
+        Intent mapIntent = new Intent(this, MapsActivity.class);
+        Request request = new Request();
```

Comment:
This is where we need to retrieve the relevant request from Firebase using book name and signed in username